internal/trace/trace_test.go

```diff
@@ -51,3 +184,110 @@ func TestToStatus(t *testing.T) {
+			"some specific grpc error",
+		},
+	} {
+		got := toOpenTelemetryStatusDescription(testcase.input)
+		if got != testcase.want {
+			t.Errorf("got %s, want %s", got, testcase.want)
+		}
+	}
+}
+
+func TestToStatus_APIError(t *testing.T) {
+	for _, testcase := range []struct {
+		input error
+		want  octrace.Status
+	}{
+		{
+			// Apparently nonsensical error, but this is supported by the implementation.
+			&googleapi.Error{Code: 200, Message: "OK"},
+			octrace.Status{Code: int32(code.Code_OK), Message: "OK"},
+		},
+		{
+			&googleapi.Error{Code: 499, Message: "error 499"},
+			octrace.Status{Code: int32(code.Code_CANCELLED), Message: "error 499"},
+		},
+		{
+			&googleapi.Error{Code: http.StatusInternalServerError, Message: "error 500"},
+			octrace.Status{Code: int32(code.Code_UNKNOWN), Message: "error 500"},
+		},
+		{
+			&googleapi.Error{Code: http.StatusBadRequest, Message: "error 400"},
+			octrace.Status{Code: int32(code.Code_INVALID_ARGUMENT), Message: "error 400"},
+		},
+		{
+			&googleapi.Error{Code: http.StatusGatewayTimeout, Message: "error 504"},
+			octrace.Status{Code: int32(code.Code_DEADLINE_EXCEEDED), Message: "error 504"},
+		},
+		{
+			&googleapi.Error{Code: http.StatusNotFound, Message: "error 404"},
+			octrace.Status{Code: int32(code.Code_NOT_FOUND), Message: "error 404"},
+		},
+		{
+			&googleapi.Error{Code: http.StatusConflict, Message: "error 409"},
+			octrace.Status{Code: int32(code.Code_ALREADY_EXISTS), Message: "error 409"},
+		},
+		{
+			&googleapi.Error{Code: http.StatusForbidden, Message: "error 403"},
+			octrace.Status{Code: int32(code.Code_PERMISSION_DENIED), Message: "error 403"},
+		},
+		{
+			&googleapi.Error{Code: http.StatusUnauthorized, Message: "error 401"},
+			octrace.Status{Code: int32(code.Code_UNAUTHENTICATED), Message: "error 401"},
+		},
+		{
+			&googleapi.Error{Code: http.StatusTooManyRequests, Message: "error 429"},
+			octrace.Status{Code: int32(code.Code_RESOURCE_EXHAUSTED), Message: "error 429"},
+		},
+		{
+			&googleapi.Error{Code: http.StatusNotImplemented, Message: "error 501"},
+			octrace.Status{Code: int32(code.Code_UNIMPLEMENTED), Message: "error 501"},
+		},
+		{
+			&googleapi.Error{Code: http.StatusServiceUnavailable, Message: "error 503"},
+			octrace.Status{Code: int32(code.Code_UNAVAILABLE), Message: "error 503"},
+		},
+		{
+			&googleapi.Error{Code: http.StatusMovedPermanently, Message: "error 301"},
+			octrace.Status{Code: int32(code.Code_UNKNOWN), Message: "error 301"},
+		},
+	} {
+		// Wrap googleapi.Error in apierror.APIError as GAPIC clients do.
+		// https://github.com/googleapis/gax-go/blob/v2.12.0/v2/invoke.go#L95
+		err, ok := apierror.FromError(testcase.input)
+		if !ok {
+			t.Fatalf("apierror.FromError failed to parse %v", testcase.input)
+		}
+		got := toStatus(err)
+		if r := testutil.Diff(got, testcase.want); r != "" {
+			t.Errorf("got -, want +:\n%s", r)
+		}
+	}
+}
+
+func attrMap() map[string]interface{} {
```

Comment:
minor nit:  maybe give this a name that makes it clearer its test/example data?  Wasn't clear when I encountered it in the test code until I sought out the function implementation.

Reply:
Updated, thanks!